Bottom: 356d1a4584125bfeba19dc27d8f728efedbaed4f
Top:    54fb756d455a6db568992b6170951633d17e31c6
Author: Yang Hongyang <yanghy@cn.fujitsu.com>
Date:   2015-09-16 20:15:57 +0800

qmp: delete qemu opts when delete an object

When delete an object, we need to delete the associated qemu opts,
otherwise, we can not add another object with the same name using
object_add.
The case happens when we start qemu with:
-object xxx,id=aa
then we delete this object with:
object_del aa
then add the object with:
object_add xxx,id=aa

QEMU will return error with Duplicate ID error.

Signed-off-by: Yang Hongyang <yanghy@cn.fujitsu.com>
Signed-off-by: Jason Wang <jasowang@redhat.com>


---

diff --git a/qmp.c b/qmp.c
index 057a7cb..64c38b6 100644
--- a/qmp.c
+++ b/qmp.c
@@ -679,6 +679,7 @@ void qmp_object_del(const char *id, Error **errp)
 {
     Object *container;
     Object *obj;
+    QemuOpts *opts;
 
     container = object_get_objects_root();
     obj = object_resolve_path_component(container, id);
@@ -692,6 +693,9 @@ void qmp_object_del(const char *id, Error **errp)
         return;
     }
     object_unparent(obj);
+
+    opts = qemu_opts_find(qemu_find_opts_err("object", NULL), id);
+    qemu_opts_del(opts);
 }
 
 MemoryDeviceInfoList *qmp_query_memory_devices(Error **errp)
